data_prep/roberta_graph_dataset.py

```diff
@@ -1,24 +1,67 @@
+import nltk
+nltk.download('punkt')
+from nltk import word_tokenize
+import torch
+from torchtext.vocab import GloVe
+from transformers import RobertaModel
+
 from data_prep.graph_dataset import GraphDataset
 
 
 class RobertaGraphDataset(GraphDataset):
     """
     Text Dataset used by the Roberta graph model.
     """
-    def __init__(self, *args, **kwargs):
-        # class super init
-        # append features to data object
-        raise NotImplementedError
+    def __init__(self, corpus, device):
+        super().__init__(corpus, device)
+        self._data.doc_features, self._data.word_features = self._generate_features(self._tokens, self._raw_texts)
 
-    def preprocess(self):
+    def _preprocess(self, raw_texts):
```

Comment:
As mentioned somewhere above, I do not see the reason to pass raw_texts, if we already have the self object and can just do `self._raw_texts` in here